database/service_instance.go

```diff
@@ -0,0 +1,563 @@
+	// Asia/Seoul, Asia/Shanghai, Asia/Singapore, Asia/Taipei, Asia/Tehran, Asia/Tokyo, Asia/Ulaanbaatar,
+	// Asia/Vladivostok, Asia/Yakutsk, Asia/Yerevan, Atlantic/Azores, Australia/Adelaide,
+	// Australia/Brisbane, Australia/Darwin, Australia/Hobart, Australia/Perth, Australia/Sydney,
+	// Brazil/East, Canada/Newfoundland, Canada/Saskatchewan, Europe/Amsterdam, Europe/Athens,
+	// Europe/Dublin, Europe/Helsinki, Europe/Istanbul, Europe/Kaliningrad, Europe/Moscow,
+	// Europe/Paris, Europe/Prague, Europe/Sarajevo, Pacific/Auckland, Pacific/Fiji, Pacific/Guam,
+	// Pacific/Honolulu, Pacific/Samoa, US/Alaska, US/Central, US/Eastern, US/East-Indiana,
+	// US/Pacific, UTC.
+	// Default value is UTC.
+	// Optional.
+	Timezone string `json:"timezone,omitempty"`
+	// Component type to which the set of parameters applies.
+	// Valid values are: db - Oracle Database
+	// Required.
+	Type ServiceInstanceType `json:"type"`
+	// Storage size for data (in GB). Minimum value is 15. Maximum value depends on the backup
+	// destination: if BOTH is specified, the maximum value is 1200; if OSS or NONE is specified,
+	// the maximum value is 2048.
+	// Required.
+	UsableStorage string `json:"usableStorage"`
+}
+
+type AdditionalParameters struct {
+	// Indicates whether to include the Demos PDB
+	// Optional
+	DBDemo ServiceInstanceBool `json:"db_demo,omitempty"`
+}
+
+// CreateServiceInstance creates a new ServiceInstace.
+func (c *ServiceInstanceClient) CreateServiceInstance(input *CreateServiceInstanceInput) (*ServiceInstance, error) {
+	var (
+		serviceInstance      *ServiceInstance
+		serviceInstanceError error
+	)
+	// return nil, fmt.Errorf("%+v", input)
+	for i := 0; i < *c.DatabaseClient.client.MaxRetries; i++ {
+		c.client.DebugLogString(fmt.Sprintf("(Iteration: %d of %d) Creating service instance with name %s\n Input: %+v", i, *c.DatabaseClient.client.MaxRetries, input.Name, input))
+
+		serviceInstance, serviceInstanceError = c.startServiceInstance(input.Name, input)
+		if serviceInstanceError == nil {
+			c.client.DebugLogString(fmt.Sprintf("(Iteration: %d of %d) Finished creating service instance with name %s\n Info: %+v", i, *c.DatabaseClient.client.MaxRetries, input.Name, serviceInstance))
+			return serviceInstance, nil
+		}
+	}
+	return nil, serviceInstanceError
+}
+
+func (c *ServiceInstanceClient) startServiceInstance(name string, input *CreateServiceInstanceInput) (*ServiceInstance, error) {
+	if err := c.createResource(*input, nil); err != nil {
+		return nil, err
+	}
+
+	// Call wait for instance ready now, as creating the instance is an eventually consistent operation
+	getInput := &GetServiceInstanceInput{
+		Name: name,
+	}
+
+	// Wait for the service instance to be running and return the result
+	// Don't have to unqualify any objects, as the GetServiceInstance method will handle that
+	serviceInstance, serviceInstanceError := c.WaitForServiceInstanceRunning(getInput, WaitForServiceInstanceReadyTimeout)
```

Comment:
Then modify this function call to use the `time.Duration` value passed in as the input.